components/ThemeComponents/ThemeContext.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react';
+import { getTheme } from './ThemeColorschemes';
+
+const ContextData = {
+  theme: '',
+  toggleTheme: () => {},
+};
+
+const ThemeContext = React.createContext(ContextData);
+const useTheme = () => React.useContext(ThemeContext);
+
+class ThemeToggleProvider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = '';
+  }
+
+  componentDidMount() {
+    this.state =
+      getComputedStyle(document.documentElement).getPropertyValue('--primary') == '#ffffff' ? 'light' : 'dark';
```

Comment:
Assigning to the `this.state` variable in class Components is not permitted in React.
To change state you have to use `this.setState`.

https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly